client/src/client.rs

```diff
@@ -351,6 +351,29 @@ pub trait RpcApi: Sized {
         self.call("getmininginfo", &[])
     }
 
+    fn get_block_template(
+        &self,
+        mode: json::GetBlockTemplateModes,
+        rules: &[json::GetBlockTemplateRules],
+        capabilities: &[json::GetBlockTemplateCapabilities],
+    ) -> Result<json::GetBlockTemplateResult> {
+        #[derive(Serialize)]
+        struct Argument {
+            mode: json::GetBlockTemplateModes,
+            rules: Vec<json::GetBlockTemplateRules>,
+            capabilities: Vec<json::GetBlockTemplateCapabilities>,
```

Comment:
There is no reason for these not to be slices as well instead of `Vec`'s. serde can simply serialize slices without an issue :)

Reply:
All you'll need to do is add a lifetime parameter `'a` to the `Argument` definition, to become `Argument<'a>`.

Reply:
Ahh, thanks for the pointer!